documents/Specification-Format-Style-Guidelines.md

```diff
@@ -0,0 +1,296 @@
+- Use blank lines (with `*` prefix) for paragraph breaks in message or
+  enum declarations, regardless of where defined.
+   - Use `<p>` tags for _fields_, and _no_ "empty" lines (it breaks the tables).
+- Use HTML tags where appropriate, but stick to markdown as much as possible.
+- Use correct US-English spelling and grammar to the maximum extent possible.
+   - Write proper and complete sentences, avoid partial sentences and
+     unnecessary abbreviations.
+   - Do not use partial words (e.g. instead of "spec" write "specification").
+   - Explain all acronyms on first use in a document.
+      - Use "Hedera Consensus Service(HCS)" the first time "HCS" is written
+        but "HCS" may be used thereafter.
+      - Extremely common acronyms (e.g. RAM or LASER) do not need
+        to be expanded.
+- First line of each document block is "what is this", and should be a single
+  sentence.  Examples: "An identifier for an account." or "A `tokenTransfer`
+  transaction request body.".
+   - Avoid "this is a thing" (or shorter "thing") type sentences,
+     write "A thing" instead.
+      - This applies for the _first sentence_ only, **not** elsewhere.
+        The first sentence is a "what is this" answer, not full description
+        and mostly just translates from "code" syntax to plain English.
+   - Minimize detail here, qualifiers, usage, etc... belong in the brief
+     description or requirements, not here.
+      - A good rule of thumb is that if this first sentence exceeds 60
+        characters, then it is trying to describe too much.
+- Include a line break (`<br/>`) after the first line, then a brief description,
+  if appropriate.
+   - Not every field needs additional description, but _every_ `message` does.
+   - Descriptions may be detailed, but should not include internal
+     implementation details or requirements. The description _should_ be
+     detailed enough to explain to an external contributor what the
+     message or field contains, where it fits within the overall network
+     system, and why it is needed.
+      - This is a long-term goal; we have a long way to go to get to this
+        level of description. Every PR to change `.proto` files should seek
+        to improve the documentation for that file.
+   - Avoid describing what must, should, must not, or should not be
+     sent, or how the field values affect output. Those items belong in the
+     _requirements_, instead.
+- After a paragraph break (a blank line for messages), one requirement per
+  line. Use break tags at the end of each requirement except the last.
+   - In field comments a `<p>` tag is _required_ for paragraph breaks.
+     Blank lines break the automated documentation generator in **fields**
+     because fields are output as markdown tables, and markdown tables require
+     each row to be a single long line.
+- Be _prescriptive_ in requirements.  Do not _describe_ the behavior.
```

Comment:
In proto specification 2 we had the `required` tag but that has been removed in v3. Should every field now try to have the equivalent of either an `optional` or `required` tag in the comments e.g. "This field SHALL not be empty" or what are the guidelines around that?

Reply:
`optional` still exists in proto3 and later versions (2023 edition is latest, I believe); it requests "explicit presence".
Required fields are the only ones we probably should describe, since optional is the default.
Note, `PBJ` currently _**does not**_ support explicit presence, so that feature gap will need to be closed before we can enforce such a guideline.

How would this read, as a guideline?
> The default state for any field is "optional", and all required fields should be specified as such.  If the presence or absence of a field must be clear _separate_ from the default value of that field, the field should use proto3 "explicit presence" and be marked with the `optional` keyword.

Reply:
Updated.